nntrainer/compiler/slice_realizer.cpp

```diff
@@ -30,20 +30,19 @@ SliceRealizer::realize(const GraphRepresentation &reference) {
     NodeInfo(std::shared_ptr<LayerNode> node) :
       node(node),
       is_visited(false),
-      is_added(false) {}
+      to_be_added(false) {}
     std::shared_ptr<LayerNode> node; /**< set this if not visited */
     bool is_visited;                 /**< set this if visited */
-    bool is_added;                   /**< set this if added */
+    bool to_be_added;                /**< set this if it is to be added */
     std::vector<std::string> children;
-    std::vector<std::string> path;
-    /**< path is the tracing result from start to current node
-      eg) if traversal has started from a -> b -> c -> d.
-      The path has {"a", "b", "c", "d"} */
 
     LayerNode *operator->() { return node.get(); }
   };
 
-  std::unordered_map<std::string, NodeInfo> mp; /// map point
+  /** @note mp has to be ordered map to keep the ordering of the nodes in the
+   * graph */
+  std::map<std::string, NodeInfo> mp; /// map point
```

Comment:
Could you elaborate on this? this will be ordered by name of node, which I think wouldn't mean anything.